dt-impact-tracer/src/main/java/edu/washington/cs/dt/impact/Main/AutoCleanerMain.java

```diff
@@ -0,0 +1,181 @@
+        super(args);
+
+        this.classpath = FailedTestRemover.buildClassPath(getArgRequired("classpath").split(":"));
+        this.cleanerTestName = getArgRequired("cleanerTestName");
+        this.victimTestName = getArgRequired("victimTestName");
+        this.testFilesPath = Paths.get(getArgRequired("tests"));
+        this.testBinaryPath = Paths.get(getArgRequired("testBinaryPath"));
+    }
+
+    private boolean didTestsPass(List<String> tests) throws IOException {
+        FailingTestDetector detector = getFailingTestDetector(classpath, tests);
+        Set<String> failedTest = detector.notPassingTests();
+        return failedTest.isEmpty();
+    }
+
+    private FailingTestDetector detector;
+    private FailingTestDetector getFailingTestDetector(String classpath, List<String> tests) {
+        if (detector == null) {
+            detector = new FailingTestDetector(classpath, new ArrayList<>(tests));
+        } else {
+            detector.setTestList(new ArrayList<>(tests));
+        }
+        return detector;
+    }
+
+    private void compile(final JavaFile javaFile) throws Exception {
+        int errorCount = -1;
+        while (errorCount != 0) {
+            System.out.println("Trying to compile: " + javaFile.getFilename());
+            final DiagnosticCollector<JavaFileObject> diagnostics = javaFile.tryCompile();
+
+            errorCount = FailedTestRemover.getDiagnosticCount(diagnostics, Diagnostic.Kind.ERROR);
+
+            System.out.println("Compiling " + (errorCount == 0 ? "succeeded" : "failed")
+                                       + " with " + errorCount + " errors.");
+            errorCount = 0;
+        }
+
+        System.out.println("Successfully compiled at: " + new Date().toString());
+        System.out.println();
+    }
+
+    private JavaFile getJavaFile(String testName, List<Path> testFiles, boolean saveOrigFile, Path outputPath) throws IOException {
+        // For each test source files, find the cleaner and victim test name
+        for (Path p : testFiles) {
+            if (!p.getFileName().toString().endsWith(".java")) {
+                continue;
+            }
+
+            Path pOrig = Paths.get(p.toString() + ".orig");
+
+            if (Files.exists(pOrig)) {
+                Files.copy(pOrig, p,
+                           StandardCopyOption.REPLACE_EXISTING);
+            } else if (saveOrigFile){
+                saveOrigFile(p);
+            }
+
+            // Copy the compiled binary to the correct location
+            // TODO a better way should be implemented to look up the location of the
+            // binary file using the source file that we found, fixed, and recompiled
+            JavaFile jFile = new JavaFile(p.getFileName().toString(), classpath, outputPath);
+
+            jFile.loadClassList(p);
+            if (jFile.getMethodDeclaration(testName) != null) {
+                return jFile;
+            }
+        }
+        return null;
```

Comment:
This method returns null, but all methods that call it assume it doesn't. Would be good to add some error checking at some point.